### PR TITLE
DAOS-3676 Test: Verify container destroy during aggregation

### PR DESCRIPTION
**Pull request by ravalsam**
Adding Yield in aggrgeation code path so the container can be destroyed
while aggregation is in progress.

Signed-off-by: Samir Raval <samir.raval@intel.com>

### CODE REVIEW

File: src/tests/suite/daos_container.c
Comment (NiuYawei) [inline src/tests/suite/daos_container.c:2233]: Do we really need to wait that long? 25 seconds isn't enough?

File: src/vos/vos_aggregate.c
Comment (NiuYawei) [inline src/vos/vos_aggregate.c:1754]: It's better mark VOS_ITER_CB_YIELD by "*act |= VOS_ITER_CB_YIELD". (though your test doesn't update VOS tree while yielding)

File: src/tests/suite/daos_container.c
Comment (NiuYawei) [inline src/tests/suite/daos_container.c:2217]: That implies the test can only run on nodes >= 2, maybe we can use OC_SX that the test can run on single node as well.

File: src/tests/suite/daos_container.c
Comment (NiuYawei) [inline src/tests/suite/daos_container.c:2227]: Maybe larger dataset to make aggregation last longer?

File: src/include/daos/common.h
Comment (ravalsam) [reply src/include/daos/common.h:680]: sure will combine.

File: src/tests/suite/daos_container.c
Comment (ravalsam) [reply src/tests/suite/daos_container.c:2217]: oh sure.

File: src/tests/suite/daos_container.c
Comment (ravalsam) [reply src/tests/suite/daos_container.c:2227]: I tested and this worked so though not to write more data but can increase

File: src/tests/suite/daos_container.c
Comment (ravalsam) [reply src/tests/suite/daos_container.c:2233]: Yes, I wanted to run top of aggregation time to be sure it's about to start. And is it 25 or 30 seconds?

File: src/vos/vos_aggregate.c
Comment (ravalsam) [reply src/vos/vos_aggregate.c:1754]: hmm ok, let me understand.

File: src/tests/suite/daos_container.c
Comment (NiuYawei) [reply src/tests/suite/daos_container.c:2227]: The test is like: 1. block aggregation; 2. unblock aggregation; 3. destroy container; so I hope the aggregation not finishing before the 3rd step. Maybe larger dataset could be helpful.

File: src/tests/suite/daos_container.c
Comment (NiuYawei) [reply src/tests/suite/daos_container.c:2233]: I think 30 seconds should be enough.

File: src/vos/vos_aggregate.c
Comment (NiuYawei) [reply src/vos/vos_aggregate.c:1754]: I mean adding a "*act |= VOS_ITER_CB_YIELD" after this ABT_thread_yield().

**Review by jolivier23 — APPROVED** [review]

**Comment by ravalsam** [comment]
There is a single test failure and looks similar to the existing defect DAOS-6367.

**Comment by kprantis** [comment]
Is this patch now just waiting for reviews?

**Comment by ravalsam** [comment]
> Is this patch now just waiting for reviews?

Yes, Niu has reviewed it initially but he is out on Holiday, so once he is back will get it reviewed.

**Review by NiuYawei — DISMISSED** [review]

File: src/tests/suite/daos_container.c
Comment (jolivier23) [inline src/tests/suite/daos_container.c:2246]: assert_rc_equal

**Review by jolivier23 — CHANGES_REQUESTED** [review]
I'd like to ensure this doesn't create instability with that

**Review by jolivier23 — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]

**Pull request merged**
